### PR TITLE
docs(mnq): Guide to set up Messaging configuration and Terraform NATS

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

**Review by quantumsheep — APPROVED** [review]

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# How to use Scaleway Messaging and Queuing config
+
+In this guide you'll learn how to deploy Scaleway Messaging and Queuing config and use the NATS Jetstream provider with,
+which is a plugin for Terraform that allows you to provision and manage NATS Jetstream resources.
+
+## Prerequisites
+
+* First, you will need to set up a new Terraform configuration file
+  with [Scaleway](https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/resources/mnq_namespace)
+  and [Jetstream providor](https://registry.terraform.io/providers/nats-io/jetstream/latest/docs/guides/setup).
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:16]:
```suggestion
  and [Jetstream provider](https://registry.terraform.io/providers/nats-io/jetstream/latest/docs/guides/setup).
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+* At this point you can configure the [NATS CLI](https://docs.nats.io/using-nats/nats-tools/nats_cli) with your
+  endpoint.
+
+```shell
+nats context save example --server nats://nats.mnq.fr-par.scw.cloud:4222 --description 'Prod.Net Server'
+```
+
+* NATS let you use `contexts` that you can store and easily select. Check more
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:55]: Maybe this goes before the previous shell snippet?

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+```shell
+NATS Configuration Context "example"
+
+      Description: Prod.Net Server
+      Server URLs: nats://nats.mnq.fr-par.scw.cloud:4222
+             Path: /Your/path/context/example.json
+```
+
+* Before to create your credentials, be aware that despite being an
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:66]:
```suggestion
* Before creating your credentials, be aware that due to an
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  ongoing [issue](https://github.com/hashicorp/terraform/issues/516) since 2014, secrets stored in `terraform.tfstate`
+  remain in plain text. While there are methods to remove secrets from state files, they are unreliable and may not
+  function properly with updates to Terraform. It is not recommended to use these workarounds.
+
+---
+  Otherwise, the official update on December3, 2020:
+  Terraform 0.14 has added the ability to mark variables as sensitive, which helps keep them out of your logs, so you should add `sensitive = true` to variables!
+---
+
+* You can create a Credential easily using the Scaleway provider.
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:76]:
```suggestion
* You can create Credentials easily using the Scaleway provider.
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  Check our example below and more about authenticating with a Credentials
+  File [here](https://docs.nats.io/using-nats/developer/connecting/creds)
+
+```hcl
+resource "scaleway_mnq_credential" "main" {
+  name         = "creed-ns"
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:82]:
```suggestion
  name         = "creds-ns"
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+```shell
+NATS Configuration Context "example"
+
+      Description: Prod.Net Server
+      Server URLs: nats://nats.mnq.fr-par.scw.cloud:4222
+      Credentials: /Your/path/secret/admin.creds (OK)
+             Path: /Your/path/context/example.json
+```
+
+* Finally configuring the provider with the server and credentials for the NATS Jetstream service.
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:103]:
```suggestion
* Finally configure the provider with the server and credentials for the NATS Jetstream service.
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+page_title: "Using Scaleway Messaging and Queuing service with NATS Terraform provider"
+description: |-
+Using Scaleway Messaging and Queuing service with NATS Terraform provider
+---
+
+# How to use Scaleway Messaging and Queuing config
+
+In this guide you'll learn how to deploy Scaleway Messaging and Queuing config and use the NATS Jetstream provider with,
+which is a plugin for Terraform that allows you to provision and manage NATS Jetstream resources.
+
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:11]: Maybe a quick recap here of what will be done during this guide

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
+---
+page_title: "Using Scaleway Messaging and Queuing service with NATS Terraform provider"
+description: |-
+Using Scaleway Messaging and Queuing service with NATS Terraform provider
+---
+
```
Comment (scwgoire) [inline docs/guides/mnq_with_nats_terraform_provider.md:6]: General remark: I find it hard to follow the constant switching between terraform and nats cli

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+# How to use Scaleway Messaging and Queuing config
+
+In this guide you'll learn how to deploy Scaleway Messaging and Queuing config and use the NATS Jetstream provider with,
```
Comment (remyleone) [inline docs/guides/mnq_with_nats_terraform_provider.md:9]: This guide shows the combination of Scaleway Messaging and Queuing configuration with the terraform NATS Jetstream provider. Il will allow you to provision and manage NATS Jetstream resources.

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+which is a plugin for Terraform that allows you to provision and manage NATS Jetstream resources.
+
+At the end of this guide you will have a running NATS server and a configured NATS CLI to interact with.
```
Comment (remyleone) [inline docs/guides/mnq_with_nats_terraform_provider.md:12]:
```suggestion
At the end of this guide, you will have a running NATS server and a configured NATS CLI to interact with it.
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+```hcl
+resource "scaleway_mnq_credential" "main" {
+  name         = "creds-ns"
+  namespace_id = scaleway_mnq_namespace.main.id
+}
+```
+
+* At this point you have your Namespace and your Credential that means you have a running NATS Server ready to be used.
+
+* Grab a copy of the  [NATS CLI](https://github.com/nats-io/jetstream/releases) and configure it with your
```
Comment (remyleone) [inline docs/guides/mnq_with_nats_terraform_provider.md:76]:
```suggestion
* Grab a copy of the [NATS CLI](https://github.com/nats-io/jetstream/releases) and configure it with your
```

File: docs/guides/mnq_with_nats_terraform_provider.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+  endpoint. To be practical lets use `contexts` that you can store and easily select. Check more
```
Comment (remyleone) [inline docs/guides/mnq_with_nats_terraform_provider.md:77]:
```suggestion
  endpoint. To be practical let's use `contexts` that you can store and easily select the relevant context. Check more
```

**Review by remyleone — APPROVED** [review]
